src/main/java/org/oscii/concordance/RemoteAlignedCorpus.java

```diff
@@ -67,9 +70,36 @@ public void read(String path, String sourceLanguage, String targetLanguage, int
   public static class PhrasalRule {
     List<String> sourceWords;
     List<String> targetWords;
+    double score = 0.0;
+
+    public PhrasalRule(String sourcePhrase, String targetPhrase, double score) {
+      if (sourcePhrase != null && targetPhrase != null) {
```

Comment:
Is it possible that `sourcePhrase` is null but `targetPhrase` is not? It seems like the `Collections.emptyList()` substitutions should happen independently for source and target. I'm going to make that change and push — please comment before I merge.